test/controllers/api/v1/account_controller_test.exs

```diff
@@ -0,0 +1,62 @@
+defmodule Habits.API.V1.AccountControllerTest do
+  use Habits.ConnCase
+
+  alias Habits.{Account, Session}
+
+  describe ".show" do
+
+    test "returns info about the current account", %{conn: conn} do
+      account = Factory.insert(:account)
+      habit = Factory.insert(:habit, account: account)
+      Factory.insert(:check_in, habit: habit)
+
+      conn =
+        conn
+        |> assign(:current_account, account)
+        |> get(api_v1_account_path(conn, :show))
+
+      assert json_response(conn, 200) == (
+        %{
+          "email" => account.email,
+          "totalCheckIns" => 1
+        }
+      )
+    end
+  end
+
+  describe ".create" do
+
+    test "creates a new account and signs in", %{conn: conn} do
+      data = %{
+        "account" => %{
+          "email" => "new@habits.ai",
+          "password" => "p4ssw0rd"
+        }
+      }
+      conn = post(conn, api_v1_account_path(conn, :create), data)
+
+      account = Repo.get_by(Account, email: "new@habits.ai")
+      assert account
+      session = Repo.get_by(Session, account_id: account.id)
+      assert session
```

Comment:
I initially wanted this to be

```elixir
assert session = Repo.get_by(Session, account_id: account.id)
```

But the compiled complained about

```
warning: this clause cannot match because a previous clause at line 42 always matches
  test/controllers/api/v1/account_controller_test.exs:42
```

I don't understand this though. `session = nil` returns `nil` which `assert` should be able to match on, but perhaps it's doing some metaprogramming behind the scenes.